misk-aws/src/main/kotlin/misk/jobqueue/sqs/SqsJobQueue.kt

```diff
@@ -17,14 +19,21 @@ import javax.inject.Singleton
 internal class SqsJobQueue @Inject internal constructor(
   private val queues: QueueResolver,
   private val metrics: SqsMetrics,
+  private val moshi: Moshi,
   private val tracer: Tracer
 ) : JobQueue {
   override fun enqueue(
     queueName: QueueName,
     body: String,
+    idempotenceKey: String,
     deliveryDelay: Duration?,
     attributes: Map<String, String>
   ) {
+    // Ensure there are at most 8 attributes; AWS SQS enforces a limit of 10 custom attributes
+    // per message, 2 of which are reserved for this library (trace id and jobqueue metadata).
+    // https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-message-attributes.html
+    check(attributes.size <= 8) { "a maximum of 8 attributes are supported (got ${attributes.size})" }
```

Comment:
the trace id should just go in the metadata

Reply:
Initially did that but didn't want to break backwards compatibility or make it harder to inspect jobs; then again, this PR is the perfect time for this change.

Will change.

Reply:
well I think you want both for the rollout. keep it a separate attribute _and_ move it into the metadata

Reply:
Updated.